src/init/update.sh

```diff
@@ -7,137 +7,374 @@
+                PREINSTALLEDDIR=`sed -n 's/^WAZUH_HOME=\(.*\)$/\1/p' /etc/rc.d/init.d/${service}`
+                if [ -d "$PREINSTALLEDDIR" ]; then
+                    return 0;
+                else
+                    return 1;
+                fi
+            else
+                return 1;
+            fi
+        # Checking for Debian (Ubuntu or derivative)
+        elif [ -d "/etc/init.d" -a -f "/usr/sbin/update-rc.d" ]; then
+            if [ -f /etc/init.d/${service} ]; then
+                PREINSTALLEDDIR=`sed -n 's/^WAZUH_HOME=\(.*\)$/\1/p' /etc/init.d/${service}`
+                if [ -d "$PREINSTALLEDDIR" ]; then
+                    return 0;
+                else
+                    return 1;
+                fi
+            else
+                return 1;
+            fi
+        fi
+    fi
+
+    return 1;
 }
 
+##########
+# Checks if Wazuh is installed by trying with each installation type. If it finds
+# an installation, it sets the PREINSTALLEDDIR variable.
+#
+# getPreinstalledDir()
+##########
 getPreinstalledDir()
 {
-    . ${OSSEC_INIT}
-    echo "$DIRECTORY"
+    # Checking ossec-init.conf for old wazuh versions
+    if [ -f "${OSSEC_INIT}" ]; then
+        . ${OSSEC_INIT}
+        if [ "X$DIRECTORY" = "X" ]; then
+            return 1;
+        fi
+        if [ -d "$DIRECTORY" ]; then
+            PREINSTALLEDDIR="$DIRECTORY"
+            return 0;
+        fi
+    else
+        # Getting preinstalled dir for Wazuh manager and Hibrid installations
+        service="wazuh-manager"
+        if getPreinstalledDirByType; then
+            return 0;
+        fi
+
+        # Getting preinstalled dir for Wazuh agent installations
+        service="wazuh-agent"
+        if getPreinstalledDirByType; then
+            return 0;
+        fi
+
+        # Getting preinstalled dir for Wazuh agent installations
+        service="wazuh-local"
+        if getPreinstalledDirByType; then
+            return 0;
+        fi
+    fi
+
+    return 1;
+}
+
+getPreinstalledType()
+{
+    # Checking ossec-init.conf for old wazuh versions
+    if [ -f "${OSSEC_INIT}" ]; then
+        . ${OSSEC_INIT}
+    else
+        if [ "X$PREINSTALLEDDIR" = "X" ]; then
+            getPreinstalledDir
+        fi
+
+        TYPE=`$PREINSTALLEDDIR/bin/wazuh-control -t`
+    fi
+
+    echo $TYPE
     return 0;
 }
 
 getPreinstalledVersion()
 {
-    . ${OSSEC_INIT}
+    # Checking ossec-init.conf for old wazuh versions
+    if [ -f "${OSSEC_INIT}" ]; then
+        . ${OSSEC_INIT}
+    else
+        if [ "X$PREINSTALLEDDIR" = "X" ]; then
+            getPreinstalledDir
+        fi
+
+        VERSION=`$PREINSTALLEDDIR/bin/wazuh-control -v`
+    fi
+
     echo $VERSION
 }
 
 getPreinstalledName()
 {
     NAME=""
-    . ${OSSEC_INIT}
+    # Checking ossec-init.conf for old wazuh versions. New versions
+    # do not provide this information at all.
+    if [ -f "${OSSEC_INIT}" ]; then
+        . ${OSSEC_INIT}
+    fi
     echo $NAME
```

Comment:
A new installation (without `ossec-init.conf`) would return an empty string.
The variable [USER_OLD_NAME](https://github.com/wazuh/wazuh/blob/ca6cb3fcabe436a3532ae5e4abb9f6905e6efdbb/install.sh#L913) wouldn't have the name "Wazuh" and the check for [old versions](https://github.com/wazuh/wazuh/blob/ca6cb3fcabe436a3532ae5e4abb9f6905e6efdbb/install.sh#L1041) would fail.
We can simple add an `else` and write **Wazuh** in case that file doesn't exist.

Reply:
Fixed